addons/mail/models/discuss/discuss_channel.py

```diff
@@ -673,6 +673,80 @@ def _add_members(
                     current_channel_member.sudo()._rtc_invite_members(member_ids=new_members.ids)
         return all_new_members
 
+    def invite_by_email(self, addresses):
+        """
+        Send channel invitation emails to a list of email addresses.
+
+        - Existing members' email addresses are ignored.
+        - An invitation can only be re-sent to the same email address for the
+        same channel after 5 minutes.
+
+        :param addresses: List of email addresses to invite.
+        :type addresses: list[str]
+
+        """
+        self.check_access("read")
+        if not self._allow_invite_by_email():
+            raise UserError(
+                self.env._("Inviting by email is not allowed for this channel type (%s).")
+                % self.channel_type
+            )
+        addresses = {
```

Comment:
Don't override input, use another variable, always easier when reading the code :)
```suggestion
        normalized_emails = {
```